### PR TITLE
chore: misc corrections

### PR DESCRIPTION
**Pull request by dlouzan**
- refactor unit tests to avoid code duplication
- fix wrong root folder to tests, shows correct code coverage of
  src files
- add minimum node 8 to README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,10 @@ the following:
 - If `legacy_mode: false` or undefined (default mode): Gitlab 11.2+
 - If `legacy_mode: true`: Gitlab 9.0+
 
+## Node Compatibility
+
+verdaccio-gitlab requires node 8+
```
Comment (bufferoverflow) [inline README.md]: we have that info below `You need at least node version 8.x.x, codename **carbon**.`

File: test/unit/authcache.spec.js [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
 
-const logger: Logger = {
-  error: jest.fn(),
-  info: jest.fn(),
-  debug: jest.fn(),
-  child: jest.fn(),
-  warn: jest.fn(),
-  http: jest.fn(),
-  trace: jest.fn()
+const TEST_USER: string = 'myUser';
```
Comment (bufferoverflow) [inline test/unit/authcache.spec.js]: this is used within both files, so maybe a common file?

File: README.md (hunk 1 above)
Comment (dlouzan) [reply README.md]: Oops yo are right :-)

File: test/unit/authcache.spec.js (hunk 2 above)
Comment (dlouzan) [reply test/unit/authcache.spec.js]: I'm going to refactor this into the `partials` objects as verdaccio does.

**Pull request merged**